gamestonk_terminal/etf/technical_analysis/ta_controller.py

```diff
@@ -0,0 +1,1351 @@
+"""Technical Analysis Controller Module"""
+__docformat__ = "numpy"
+# pylint:disable=too-many-lines
+# pylint:disable=R0904,C0201
+
+import argparse
+import difflib
+from typing import List, Union
+from datetime import datetime
+
+import pandas as pd
+from prompt_toolkit.completion import NestedCompleter
+
+from gamestonk_terminal import feature_flags as gtff
+from gamestonk_terminal.helper_funcs import (
+    EXPORT_BOTH_RAW_DATA_AND_FIGURES,
+    get_flair,
+    parse_known_args_and_warn,
+    check_positive_list,
+    check_positive,
+    try_except,
+    valid_date,
+    system_clear,
+)
+from gamestonk_terminal.menu import session
+from gamestonk_terminal.common.technical_analysis import (
+    custom_indicators_view,
+    momentum_view,
+    overlap_model,
+    overlap_view,
+    trend_indicators_view,
+    volatility_model,
+    volatility_view,
+    volume_view,
+)
+
+
+class TechnicalAnalysisController:
+    """Technical Analysis Controller class"""
+
+    # Command choices
+    CHOICES = [
+        "cls",
+        "home",
+        "h",
+        "?",
+        "help",
+        "q",
+        "quit",
+        "..",
+        "exit",
+        "r",
+        "reset",
+    ]
+
+    CHOICES_COMMANDS = [
+        "load",
+        "ema",
+        "sma",
+        "wma",
+        "hma",
+        "zlma",
+        "cci",
+        "macd",
+        "rsi",
+        "stoch",
+        "fisher",
+        "cg",
+        "adx",
+        "aroon",
+        "bbands",
+        "donchian",
+        "kc",
+        "ad",
+        "adosc",
+        "obv",
+        "fib",
+    ]
+
+    CHOICES += CHOICES_COMMANDS
+
+    def __init__(
+        self,
+        ticker: str,
+        start: datetime,
+        etf: pd.DataFrame,
+        queue: List[str] = None,
+    ):
+        """Constructor"""
+        self.ticker = ticker
+        self.start = start
+        self.etf = etf
+
+        self.ta_parser = argparse.ArgumentParser(add_help=False, prog="ta")
+        self.ta_parser.add_argument(
+            "cmd",
+            choices=self.CHOICES,
+        )
+        self.completer: Union[None, NestedCompleter] = None
+        if session and gtff.USE_PROMPT_TOOLKIT:
+            choices: dict = {c: {} for c in self.CHOICES}
+            self.completer = NestedCompleter.from_nested_dict(choices)
+
+        if queue:
+            self.queue = queue
+        else:
+            self.queue = list()
+
+    def print_help(self):
+        """Print help"""
+        help_str = f"""
+
+ETF: {self.etf}
```

Comment:
here is the rogue print.  I would think of naming this self.data instead of self.etf.  Looking at self.etf I would expect it to be the ticker.

Reply:
Gotcha